pkg/reporter/namespace_report.go

```diff
@@ -225,25 +229,31 @@ func (n *NamespaceReportHandler) getResourceFromProvider(ctx context.Context, pr
 		return handleProviderError(logger, obj, errors.Errorf("Failed request responded with status %d body %s", resp.StatusCode, string(body)), "request")
 	}
 
-	providerResp := ProviderResponse{}
-	err = json.Unmarshal(body, &providerResp)
-	if err != nil {
-		return handleProviderError(logger, obj, err, "unmarshal json")
+	providerOpsResp := ProviderOpsResponse{}
+	err = json.Unmarshal(body, &providerOpsResp)
+	if err != nil || providerOpsResp.Result == nil {
```

Comment:
I don't see how `err != nil` can be caused by `Provider using old report spec`? These are distinct cases and I think should be handled separately in code.